src/store/index.ts

```diff
@@ -80,41 +90,101 @@ export default createStore<State>({
         payload: messageData.ciphertext
       });
     },
-    async uploadFile({ dispatch }, request: UploadRequest) {
+    async uploadFile({ dispatch, getters, state }, request: UploadRequest) {
       const res = await axios.post<UploadResponse>(
         "https://rtc-data.registree.io/data/uploads",
         request
       );
       if (res.status !== 200) {
+        // TODO: Error Handling
         return "error";
       }
-      await dispatch("decryptUploadResponse", res.data);
-    },
-    async decryptUploadResponse(
-      { state, dispatch, getters },
-      response: UploadResponse
-    ) {
+
+      const { nonce, ciphertext } = res.data;
       const enclavePubKey = getters.enclavePublicKey;
-      if (!enclavePubKey || !state.ourSecretKey) {
+      const ourSecretKey = state.ourSecretKey;
+
+      const msg = decryptMessage(
+        ciphertext,
+        nonce,
+        enclavePubKey,
+        ourSecretKey as Base64
+      );
+
+      if (!msg) {
+        // TODO: Error Handling
+        return "error";
+      }
+
+      await dispatch("parseUploadMessage", msg);
+    },
+    async postAccessForm({ commit, getters, state }, request: UploadRequest) {
+      const res = await axios.post<UploadResponse>(
+        "https://rtc-data.registree.io/auth/tokens",
+        request
+      );
+      if (res.status !== 200) {
+        // TODO: Error Handling
         return "error";
       }
+
+      const { nonce, ciphertext } = res.data;
+      const enclavePubKey = getters.enclavePublicKey;
+      const ourSecretKey = state.ourSecretKey;
+
       const msg = decryptMessage(
-        response.ciphertext,
-        response.nonce,
+        ciphertext,
+        nonce,
         enclavePubKey,
-        state.ourSecretKey
+        ourSecretKey as Base64
```

Comment:
Same as above, this cast should not be necessary if we check for null.

We should check what to do with the different cases where the `ourSecretKey` is `null`. In this case it seems like a clear exception, something that should not be possible. In other cases we might just want to restart with the attestation process.